lib/src/arithmetic/multiplier.dart

```diff
@@ -15,23 +15,53 @@ import 'package:rohd_hcl/src/arithmetic/partial_product_sign_extend.dart';
 
 /// An abstract class for all multiplier implementations.
 abstract class Multiplier extends Module {
-  /// The input to the multiplier pin [a].
+  /// The multiplicand input [a].
   @protected
   late final Logic a;
 
-  /// The input to the multiplier pin [b].
+  /// The multiplier input [b].
   @protected
   late final Logic b;
 
-  /// The multiplier treats operands and output as signed
-  bool signed;
+  /// The multiplier treats input [a] always as a signed input.
+  bool signedMultiplicand;
+
+  /// The multiplier treats input [b] always as a signed input.
+  bool signedMultiplier;
+
+  /// If not null, use this signal to select between signed and unsigned
+  /// multiplicand [a].
+  late Logic? selectSignedMultiplicand;
+
+  /// If not null, use this signal to select between signed and unsigned
+  /// multiplier [b]
+  late Logic? selectSignedMultiplier;
 
   /// The multiplication results of the multiplier.
   Logic get product;
```

Comment:
Is the product signed or unsigned?  Clarify in the API doc?

Reply:
I decided to extend the API to compute this value of signed or unsigned since it is not easy to do with a mix of boolean and null/non-null selectors.  I think there is probably some functional way to map/reduce this logic to a oneliner, but this is what I have:
```
   addOutput('isAccumulateSigned') <=
        (signedMultiplicand | signedMultiplier | signedAddend
                ? Const(1)
                : Const(0)) |
            ((selectSignedMultiplicand != null)
                ? selectSignedMultiplicand
                : Const(0)) |
            ((selectSignedMultiplier != null)
                ? selectSignedMultiplier
                : Const(0)) |
            ((selectSignedAddend != null) ? selectSignedAddend : Const(0));
```